### DISCUSSION TITLE
Table 22, Panel 3

### DISCUSSION THREAD

**Discussion by pibion** [cssi-pi-community/2022-meeting#80] [Panel 3 roundtable (Tuesday morning)]
Welcome table 22!

**Comment by pibion** [comment]
@rapier1 and I would like to point out that open source projects have a hard time getting self-sustaining community involvement!

**Comment by pibion** [comment]
Fascinating example: someone had CSSI funding for modifications to the Linux kernel and at the end of the project the code was finally strong enough to submit!  And then they had no more funding.

So it was a real struggle to make that happen.

**Comment by pibion** [comment]
We don't think every project needs to be maintained.  Question at our table: how do you measure what should be maintained?  It's a different metric(s) than new, novel projects.

**Reply by pibion** [reply]
* how many people does it impact?  How many other projects does it impact?
* https://xkcd.com/2347/
*

**Reply by pibion** [reply]
* how much does useful maintaining cost?
* how feasible is the path to true community integration?  For example, getting code into the linux kernel means that it becomes part of the linux foundation, that should review very strongly

**Comment by pibion** [comment]
Going commercial can pose real problems to the community.  Example: Globus.  We all rely on globus, but FNAL can only support it for one more year.  Decisions to have that license is a national lab decision and individual PIs can't write it into their grants.

Contributing money on a sliding scale for software maintenance feels feasible, but not fixed license fees.

**Comment by pibion** [comment]
We would all like to affirm that being on soft money absolutely blows.

**Comment by pibion** [comment]
At PSC (Pittsburgh Supercomputing Center), only admin is on hard money.  Everyone (!) down to the network engineers is on soft money.

Nonetheless, they have a really low turnover rate.  Part of this is very careful hiring and careful running of the center so that people are assigned to new projects and have some job security.

**Reply by pibion** [reply]
Having high turnover makes maintenance much more difficult - it takes time to learn a new codebase.

**Comment by pibion** [comment]
Everyone at the table would like to use Rust, but none of us have had the opportunity to spend time on it :(

**Reply by pibion** [reply]
We all are excited by the Rust community, the Python community.

OpenSSL is a wonderful and strong community, but there's still not enough people to review code.  I feel like these types of projects would really benefit from occasional federal funding!

**Reply by pibion** [reply]
Github has started a "financial contributor" to your project https://github.com/sponsors.

Could we do something like this in grant proposals?  PIs need to allocate some amount to software infrastructure.  Communities could get together and decide on how much they want to require - and then projects that don't allocate that, or provide good explanation for why they aren't allocating for it are turned down.

**Reply by pibion** [reply]
The money would need to go to a place that accepts proposals for funding, we don't think people would be comfortable with a path to non-reviewed funding.

**Comment by pibion** [comment]
RSEs need to be faculty so they can have grad students!!!!!

**Reply by rapier1** [reply]
To clarify, RSE shouldn't required to be faculty but the position should be elevated to the where they are functionally equivalent.
